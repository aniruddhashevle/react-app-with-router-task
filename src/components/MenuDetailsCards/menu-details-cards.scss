@import '../assets/styles/variables.scss';
@import '../assets/styles/mixins.scss';

.menu-details-wrapper {
    .card-wrapper {
        margin-bottom: 30px;
        &:first-of-type { margin-top: 10px; }
        &:last-of-type { margin-bottom: 0; }
        .card-content {
            .details-title {
                margin: 5px 0 24px;
                @include font-properties($font-size: 16px, $color: $brandLogoColor);
            }
            .details-data {
                margin: 0;
                @include font-properties($font-size: 18px, $line-height: 30px, $color: $blackColor);
            }
        }
        .card-action {
            justify-content: flex-end;
            .card-action-button {
                @include font-properties($font-size: 16px, $color: $brandLogoColor);
            }
        }
    }
}