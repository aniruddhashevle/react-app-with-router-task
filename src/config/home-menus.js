export const HOME_MENUS = [
    {
        id: 1,
        title: 'Sales',
        subTitle: 'Insights about Sales',
        route: '/sales',
        component: 'Sales',
        showDivider: true
    },
    {
        id: 2,
        title: 'Customers',
        subTitle: 'Insights about Customers',
        route: '/customers',
        component: 'Customers',
        showDivider: true
    },
    {
        id: 3,
        title: 'Bulletin',
        subTitle: 'Daily summary of your business',
        route: '/#',
        component: null,
        showDivider: true
    },
    {
        id: 4,
        title: 'Ask',
        subTitle: 'Ask a question',
        route: '/#',
        component: null,
        showDivider: true
    },
    {
        id: 5,
        title: 'Task',
        subTitle: 'Tools for your business',
        route: '/#',
        component: null,
        showDivider: true
    },
    {
        id: 6,
        title: 'Connect',
        subTitle: 'Connect Us',
        route: '/#',
        component: null,
        showDivider: false
    },
]