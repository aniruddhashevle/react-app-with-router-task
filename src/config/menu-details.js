export const SALES_DETAILS = [
    {
        id: 1,
        title: 'Total Sales in November?',
        details: 'You have sales worth SGD 80,586.54 in last 30 days.',
        pinned: true,
    },
    {
        id: 2,
        title: 'Total Sales this week',
        details: 'You have not sold anything this week or you have not uploaded your invoices.',
        pinned: false,
    }
]

export const CUSTOMER_DETAILS = [
    {
        id: 1,
        title: 'Total Customers in November?',
        details: 'You have 30 customers in last 30 days.',
        pinned: true,
    },
    {
        id: 2,
        title: 'Total Customers this week',
        details: 'You have 5 customers in this week.',
        pinned: true,
    }
]