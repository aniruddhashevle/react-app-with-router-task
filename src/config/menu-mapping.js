import Sales from "../components/Sales";
import Customers from "../components/Customers";

const MENU_MAPPING = {};

MENU_MAPPING['Sales'] = Sales;
MENU_MAPPING['Customers'] = Customers;

export default MENU_MAPPING;