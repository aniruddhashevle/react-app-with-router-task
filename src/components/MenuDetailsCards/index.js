import React from 'react';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import { Divider } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import './menu-details-cards.scss';

const MenuDetailsCards = ({
    config: {
        title,
        details,
        pinned
    }
}) =>
    <Card className='card-wrapper'>
        <CardContent className="card-content">
            <h4 className="details-title">{title}</h4>
            <p className="details-data">{details}</p>
        </CardContent>
        <Divider />
        <CardActions className="card-action">
            <Button className="card-action-button" color={pinned ? "primary" : "secondary"}>
                {
                    pinned ? 'PIN IT' : 'UNPIN IT'
                }
            </Button>
        </CardActions>
    </Card>

export default MenuDetailsCards;
