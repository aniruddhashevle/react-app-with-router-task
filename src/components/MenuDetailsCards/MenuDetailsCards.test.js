import React from 'react';
import MenuDetailsCards from '../MenuDetailsCards';
import { SALES_DETAILS } from '../../config/menu-details';

it('MenuDetailsCards renders correctly', () => {
    const wrapper = shallow(
        <MenuDetailsCards key={1} config={SALES_DETAILS} />
    )
    expect(wrapper).toMatchSnapshot();
});