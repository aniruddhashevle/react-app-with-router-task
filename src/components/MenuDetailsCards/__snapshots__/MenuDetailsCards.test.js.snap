// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuDetailsCards renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuDetailsCards
    config={
      Array [
        Object {
          "details": "You have sales worth SGD 80,586.54 in last 30 days.",
          "id": 1,
          "pinned": true,
          "title": "Total Sales in November?",
        },
        Object {
          "details": "You have not sold anything this week or you have not uploaded your invoices.",
          "id": 2,
          "pinned": false,
          "title": "Total Sales this week",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(CardContent)
          className="card-content"
        >
          <h4
            className="details-title"
          />
          <p
            className="details-data"
          />
        </WithStyles(CardContent)>,
        <WithStyles(Divider) />,
        <WithStyles(CardActions)
          className="card-action"
        >
          <WithStyles(Button)
            className="card-action-button"
            color="secondary"
          >
            UNPIN IT
          </WithStyles(Button)>
        </WithStyles(CardActions)>,
      ],
      "className": "card-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h4
              className="details-title"
            />,
            <p
              className="details-data"
            />,
          ],
          "className": "card-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "details-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "details-data",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Button)
            className="card-action-button"
            color="secondary"
          >
            UNPIN IT
          </WithStyles(Button)>,
          "className": "card-action",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "UNPIN IT",
            "className": "card-action-button",
            "color": "secondary",
          },
          "ref": null,
          "rendered": "UNPIN IT",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardContent)
            className="card-content"
          >
            <h4
              className="details-title"
            />
            <p
              className="details-data"
            />
          </WithStyles(CardContent)>,
          <WithStyles(Divider) />,
          <WithStyles(CardActions)
            className="card-action"
          >
            <WithStyles(Button)
              className="card-action-button"
              color="secondary"
            >
              UNPIN IT
            </WithStyles(Button)>
          </WithStyles(CardActions)>,
        ],
        "className": "card-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h4
                className="details-title"
              />,
              <p
                className="details-data"
              />,
            ],
            "className": "card-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "details-title",
              },
              "ref": null,
              "rendered": null,
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "details-data",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              className="card-action-button"
              color="secondary"
            >
              UNPIN IT
            </WithStyles(Button)>,
            "className": "card-action",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "UNPIN IT",
              "className": "card-action-button",
              "color": "secondary",
            },
            "ref": null,
            "rendered": "UNPIN IT",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Sales renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sales />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "function",
    "props": Object {
      "config": Object {
        "details": "You have sales worth SGD 80,586.54 in last 30 days.",
        "id": 1,
        "pinned": true,
        "title": "Total Sales in November?",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "function",
      "props": Object {
        "config": Object {
          "details": "You have sales worth SGD 80,586.54 in last 30 days.",
          "id": 1,
          "pinned": true,
          "title": "Total Sales in November?",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "2",
      "nodeType": "function",
      "props": Object {
        "config": Object {
          "details": "You have not sold anything this week or you have not uploaded your invoices.",
          "id": 2,
          "pinned": false,
          "title": "Total Sales this week",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
